.github/workflows/ci.yml

```diff
@@ -17,43 +15,51 @@ env:
   CARGO_TERM_COLOR: always
 
 jobs:
+  formatting:
+      name: Format code
+      runs-on: ubuntu-latest
+      steps:
+        - uses: eneoli/flakeshot@github-actions-v1
+        - name: Check formatting
+          if: github.event.pull_request.head.repo.full_name != github.repository
+          run: cargo fmt --check --verbose
+        - name: Format code
+          id: format_code
+          if: github.event.pull_request.head.repo.full_name == github.repository
+          run: |
+            cargo fmt --verbose
+            echo "modified=$(if git diff-index --quiet HEAD --; then echo "false"; else echo "true"; fi)" >> $GITHUB_OUTPUT
+
+        - name: Committing reformatted code
+          if: |
+            steps.format_code.outputs.modified == 'true' &&
+            github.event.pull_request.head.repo.full_name == github.repository
+          run: |
+            git config --global user.name 'Github Actions'
+            git config --global user.email '<>'
+            git remote set-url origin https://x-access-token:${{ secrets.GITHUB_TOKEN }}@github.com/${{ github.repository }}
+            git commit -am "Automated formatting"
+            git push
   check:
-    if: github.event.pull_request.draft == false
```

Comment:
This will trigger a check on every push. This might kill our contingent of ci minutes quickly without a huge benefit as a draft PR is likely still work in progress.

Reply:
oh good point

Reply:
fixed